// obsidian-callouts-markdown/dist/obsidian-callouts-markdown.js
import * as o from "react";
import Ne, { isValidElement as z } from "react";
var ie = { exports: {} }, W = {};
/**
 * @license React
 * react-jsx-runtime.production.min.js
 *
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var Ae;
function kr() {
  if (Ae)
    return W;
  Ae = 1;
  var n = Ne, f = Symbol.for("react.element"), y = Symbol.for("react.fragment"), p = Object.prototype.hasOwnProperty, b = n.__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED.ReactCurrentOwner, k = { key: !0, ref: !0, __self: !0, __source: !0 };
  function w(E, g, F) {
    var C, _ = {}, S = null, N = null;
    F !== void 0 && (S = "" + F), g.key !== void 0 && (S = "" + g.key), g.ref !== void 0 && (N = g.ref);
    for (C in g)
      p.call(g, C) && !k.hasOwnProperty(C) && (_[C] = g[C]);
    if (E && E.defaultProps)
      for (C in g = E.defaultProps, g)
        _[C] === void 0 && (_[C] = g[C]);
    return { $$typeof: f, type: E, key: S, ref: N, props: _, _owner: b.current };
  }
  return W.Fragment = y, W.jsx = w, W.jsxs = w, W;
}
var I = {};
/**
 * @license React
 * react-jsx-runtime.development.js
 *
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var De;
function Rr() {
  return De || (De = 1, process.env.NODE_ENV !== "production" && function() {
    var n = Ne, f = Symbol.for("react.element"), y = Symbol.for("react.portal"), p = Symbol.for("react.fragment"), b = Symbol.for("react.strict_mode"), k = Symbol.for("react.profiler"), w = Symbol.for("react.provider"), E = Symbol.for("react.context"), g = Symbol.for("react.forward_ref"), F = Symbol.for("react.suspense"), C = Symbol.for("react.suspense_list"), _ = Symbol.for("react.memo"), S = Symbol.for("react.lazy"), N = Symbol.for("react.offscreen"), le = Symbol.iterator, Ye = "@@iterator";
    function $e(e) {
      if (e === null || typeof e != "object")
        return null;
      var r = le && e[le] || e[Ye];
      return typeof r == "function" ? r : null;
    }
    var P = n.__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED;
    function h(e) {
      {
        for (var r = arguments.length, t = new Array(r > 1 ? r - 1 : 0), a = 1; a < r; a++)
          t[a - 1] = arguments[a];
        Ue("error", e, t);
      }
    }
    function Ue(e, r, t) {
      {
        var a = P.ReactDebugCurrentFrame, l = a.getStackAddendum();
        l !== "" && (r += "%s", t = t.concat([l]));
        var s = t.map(function(c) {
          return String(c);
        });
        s.unshift("Warning: " + r), Function.prototype.apply.call(console[e], console, s);
      }
    }
    var qe = !1, ze = !1, Je = !1, Ge = !1, He = !1, se;
    se = Symbol.for("react.module.reference");
    function Ke(e) {
      return !!(typeof e == "string" || typeof e == "function" || e === p || e === k || He || e === b || e === F || e === C || Ge || e === N || qe || ze || Je || typeof e == "object" && e !== null && (e.$$typeof === S || e.$$typeof === _ || e.$$typeof === w || e.$$typeof === E || e.$$typeof === g || // This needs to include all possible module reference object
      // types supported by any Flight configuration anywhere since
      // we don't know which Flight build this will end up being used
      // with.
      e.$$typeof === se || e.getModuleId !== void 0));
    }
    function Xe(e, r, t) {
      var a = e.displayName;
      if (a)
        return a;
      var l = r.displayName || r.name || "";
      return l !== "" ? t + "(" + l + ")" : t;
    }
    function ue(e) {
      return e.displayName || "Context";
    }
    function x(e) {
      if (e == null)
        return null;
      if (typeof e.tag == "number" && h("Received an unexpected object in getComponentNameFromType(). This is likely a bug in React. Please file an issue."), typeof e == "function")
        return e.displayName || e.name || null;
      if (typeof e == "string")
        return e;
      switch (e) {
        case p:
          return "Fragment";
        case y:
          return "Portal";
        case k:
          return "Profiler";
        case b:
          return "StrictMode";
        case F:
          return "Suspense";
        case C:
          return "SuspenseList";
      }
      if (typeof e == "object")
        switch (e.$$typeof) {
          case E:
            var r = e;
            return ue(r) + ".Consumer";
          case w:
            var t = e;
            return ue(t._context) + ".Provider";
          case g:
            return Xe(e, e.render, "ForwardRef");
          case _:
            var a = e.displayName || null;
            return a !== null ? a : x(e.type) || "Memo";
          case S: {
            var l = e, s = l._payload, c = l._init;
            try {
              return x(c(s));
            } catch {
              return null;
            }
          }
        }
      return null;
    }
    var j = Object.assign, L = 0, fe, de, ve, pe, Ee, ge, he;
    function me() {
    }
    me.__reactDisabledLog = !0;
    function Ze() {
      {
        if (L === 0) {
          fe = console.log, de = console.info, ve = console.warn, pe = console.error, Ee = console.group, ge = console.groupCollapsed, he = console.groupEnd;
          var e = {
            configurable: !0,
            enumerable: !0,
            value: me,
            writable: !0
          };
          Object.defineProperties(console, {
            info: e,
            log: e,
            warn: e,
            error: e,
            group: e,
            groupCollapsed: e,
            groupEnd: e
          });
        }
        L++;
      }
    }
    function Qe() {
      {
        if (L--, L === 0) {
          var e = {
            configurable: !0,
            enumerable: !0,
            writable: !0
          };
          Object.defineProperties(console, {
            log: j({}, e, {
              value: fe
            }),
            info: j({}, e, {
              value: de
            }),
            warn: j({}, e, {
              value: ve
            }),
            error: j({}, e, {
              value: pe
            }),
            group: j({}, e, {
              value: Ee
            }),
            groupCollapsed: j({}, e, {
              value: ge
            }),
            groupEnd: j({}, e, {
              value: he
            })
          });
        }
        L < 0 && h("disabledDepth fell below zero. This is a bug in React. Please file an issue.");
      }
    }
    var J = P.ReactCurrentDispatcher, G;
    function V(e, r, t) {
      {
        if (G === void 0)
          try {
            throw Error();
          } catch (l) {
            var a = l.stack.trim().match(/\n( *(at )?)/);
            G = a && a[1] || "";
          }
        return `
` + G + e;
      }
    }
    var H = !1, Y;
    {
      var er = typeof WeakMap == "function" ? WeakMap : Map;
      Y = new er();
    }
    function ye(e, r) {
      if (!e || H)
        return "";
      {
        var t = Y.get(e);
        if (t !== void 0)
          return t;
      }
      var a;
      H = !0;
      var l = Error.prepareStackTrace;
      Error.prepareStackTrace = void 0;
      var s;
      s = J.current, J.current = null, Ze();
      try {
        if (r) {
          var c = function() {
            throw Error();
          };
          if (Object.defineProperty(c.prototype, "props", {
            set: function() {
              throw Error();
            }
          }), typeof Reflect == "object" && Reflect.construct) {
            try {
              Reflect.construct(c, []);
            } catch (T) {
              a = T;
            }
            Reflect.construct(e, [], c);
          } else {
            try {
              c.call();
            } catch (T) {
              a = T;
            }
            e.call(c.prototype);
          }
        } else {
          try {
            throw Error();
          } catch (T) {
            a = T;
          }
          e();
        }
      } catch (T) {
        if (T && a && typeof T.stack == "string") {
          for (var i = T.stack.split(`
`), m = a.stack.split(`
`), d = i.length - 1, v = m.length - 1; d >= 1 && v >= 0 && i[d] !== m[v]; )
            v--;
          for (; d >= 1 && v >= 0; d--, v--)
            if (i[d] !== m[v]) {
              if (d !== 1 || v !== 1)
                do
                  if (d--, v--, v < 0 || i[d] !== m[v]) {
                    var R = `
` + i[d].replace(" at new ", " at ");
                    return e.displayName && R.includes("<anonymous>") && (R = R.replace("<anonymous>", e.displayName)), typeof e == "function" && Y.set(e, R), R;
                  }
                while (d >= 1 && v >= 0);
              break;
            }
        }
      } finally {
        H = !1, J.current = s, Qe(), Error.prepareStackTrace = l;
      }
      var D = e ? e.displayName || e.name : "", Pe = D ? V(D) : "";
      return typeof e == "function" && Y.set(e, Pe), Pe;
    }
    function rr(e, r, t) {
      return ye(e, !1);
    }
    function tr(e) {
      var r = e.prototype;
      return !!(r && r.isReactComponent);
    }
    function $(e, r, t) {
      if (e == null)
        return "";
      if (typeof e == "function")
        return ye(e, tr(e));
      if (typeof e == "string")
        return V(e);
      switch (e) {
        case F:
          return V("Suspense");
        case C:
          return V("SuspenseList");
      }
      if (typeof e == "object")
        switch (e.$$typeof) {
          case g:
            return rr(e.render);
          case _:
            return $(e.type, r, t);
          case S: {
            var a = e, l = a._payload, s = a._init;
            try {
              return $(s(l), r, t);
            } catch {
            }
          }
        }
      return "";
    }
    var U = Object.prototype.hasOwnProperty, be = {}, we = P.ReactDebugCurrentFrame;
    function q(e) {
      if (e) {
        var r = e._owner, t = $(e.type, e._source, r ? r.type : null);
        we.setExtraStackFrame(t);
      } else
        we.setExtraStackFrame(null);
    }
    function nr(e, r, t, a, l) {
      {
        var s = Function.call.bind(U);
        for (var c in e)
          if (s(e, c)) {
            var i = void 0;
            try {
              if (typeof e[c] != "function") {
                var m = Error((a || "React class") + ": " + t + " type `" + c + "` is invalid; it must be a function, usually from the `prop-types` package, but received `" + typeof e[c] + "`.This often happens because of typos such as `PropTypes.function` instead of `PropTypes.func`.");
                throw m.name = "Invariant Violation", m;
              }
              i = e[c](r, c, a, t, null, "SECRET_DO_NOT_PASS_THIS_OR_YOU_WILL_BE_FIRED");
            } catch (d) {
              i = d;
            }
            i && !(i instanceof Error) && (q(l), h("%s: type specification of %s `%s` is invalid; the type checker function must return `null` or an `Error` but returned a %s. You may have forgotten to pass an argument to the type checker creator (arrayOf, instanceOf, objectOf, oneOf, oneOfType, and shape all require an argument).", a || "React class", t, c, typeof i), q(null)), i instanceof Error && !(i.message in be) && (be[i.message] = !0, q(l), h("Failed %s type: %s", t, i.message), q(null));
          }
      }
    }
    var or = Array.isArray;
    function K(e) {
      return or(e);
    }
    function ar(e) {
      {
        var r = typeof Symbol == "function" && Symbol.toStringTag, t = r && e[Symbol.toStringTag] || e.constructor.name || "Object";
        return t;
      }
    }
    function ir(e) {
      try {
        return Ce(e), !1;
      } catch {
        return !0;
      }
    }
    function Ce(e) {
      return "" + e;
    }
    function ke(e) {
      if (ir(e))
        return h("The provided key is an unsupported type %s. This value must be coerced to a string before before using it here.", ar(e)), Ce(e);
    }
    var M = P.ReactCurrentOwner, cr = {
      key: !0,
      ref: !0,
      __self: !0,
      __source: !0
    }, Re, _e, X;
    X = {};
    function lr(e) {
      if (U.call(e, "ref")) {
        var r = Object.getOwnPropertyDescriptor(e, "ref").get;
        if (r && r.isReactWarning)
          return !1;
      }
      return e.ref !== void 0;
    }
    function sr(e) {
      if (U.call(e, "key")) {
        var r = Object.getOwnPropertyDescriptor(e, "key").get;
        if (r && r.isReactWarning)
          return !1;
      }
      return e.key !== void 0;
    }
    function ur(e, r) {
      if (typeof e.ref == "string" && M.current && r && M.current.stateNode !== r) {
        var t = x(M.current.type);
        X[t] || (h('Component "%s" contains the string ref "%s". Support for string refs will be removed in a future major release. This case cannot be automatically converted to an arrow function. We ask you to manually fix this case by using useRef() or createRef() instead. Learn more about using refs safely here: https://reactjs.org/link/strict-mode-string-ref', x(M.current.type), e.ref), X[t] = !0);
      }
    }
    function fr(e, r) {
      {
        var t = function() {
          Re || (Re = !0, h("%s: `key` is not a prop. Trying to access it will result in `undefined` being returned. If you need to access the same value within the child component, you should pass it as a different prop. (https://reactjs.org/link/special-props)", r));
        };
        t.isReactWarning = !0, Object.defineProperty(e, "key", {
          get: t,
          configurable: !0
        });
      }
    }
    function dr(e, r) {
      {
        var t = function() {
          _e || (_e = !0, h("%s: `ref` is not a prop. Trying to access it will result in `undefined` being returned. If you need to access the same value within the child component, you should pass it as a different prop. (https://reactjs.org/link/special-props)", r));
        };
        t.isReactWarning = !0, Object.defineProperty(e, "ref", {
          get: t,
          configurable: !0
        });
      }
    }
    var vr = function(e, r, t, a, l, s, c) {
      var i = {
        // This tag allows us to uniquely identify this as a React Element
        $$typeof: f,
        // Built-in properties that belong on the element
        type: e,
        key: r,
        ref: t,
        props: c,
        // Record the component responsible for creating this element.
        _owner: s
      };
      return i._store = {}, Object.defineProperty(i._store, "validated", {
        configurable: !1,
        enumerable: !1,
        writable: !0,
        value: !1
      }), Object.defineProperty(i, "_self", {
        configurable: !1,
        enumerable: !1,
        writable: !1,
        value: a
      }), Object.defineProperty(i, "_source", {
        configurable: !1,
        enumerable: !1,
        writable: !1,
        value: l
      }), Object.freeze && (Object.freeze(i.props), Object.freeze(i)), i;
    };
    function pr(e, r, t, a, l) {
      {
        var s, c = {}, i = null, m = null;
        t !== void 0 && (ke(t), i = "" + t), sr(r) && (ke(r.key), i = "" + r.key), lr(r) && (m = r.ref, ur(r, l));
        for (s in r)
          U.call(r, s) && !cr.hasOwnProperty(s) && (c[s] = r[s]);
        if (e && e.defaultProps) {
          var d = e.defaultProps;
          for (s in d)
            c[s] === void 0 && (c[s] = d[s]);
        }
        if (i || m) {
          var v = typeof e == "function" ? e.displayName || e.name || "Unknown" : e;
          i && fr(c, v), m && dr(c, v);
        }
        return vr(e, i, m, l, a, M.current, c);
      }
    }
    var Z = P.ReactCurrentOwner, xe = P.ReactDebugCurrentFrame;
    function A(e) {
      if (e) {
        var r = e._owner, t = $(e.type, e._source, r ? r.type : null);
        xe.setExtraStackFrame(t);
      } else
        xe.setExtraStackFrame(null);
    }
    var Q;
    Q = !1;
    function ee(e) {
      return typeof e == "object" && e !== null && e.$$typeof === f;
    }
    function Te() {
      {
        if (Z.current) {
          var e = x(Z.current.type);
          if (e)
            return `

Check the render method of \`` + e + "`.";
        }
        return "";
      }
    }
    function Er(e) {
      {
        if (e !== void 0) {
          var r = e.fileName.replace(/^.*[\\\/]/, ""), t = e.lineNumber;
          return `

Check your code at ` + r + ":" + t + ".";
        }
        return "";
      }
    }
    var Se = {};
    function gr(e) {
      {
        var r = Te();
        if (!r) {
          var t = typeof e == "string" ? e : e.displayName || e.name;
          t && (r = `

Check the top-level render call using <` + t + ">.");
        }
        return r;
      }
    }
    function je(e, r) {
      {
        if (!e._store || e._store.validated || e.key != null)
          return;
        e._store.validated = !0;
        var t = gr(r);
        if (Se[t])
          return;
        Se[t] = !0;
        var a = "";
        e && e._owner && e._owner !== Z.current && (a = " It was passed a child from " + x(e._owner.type) + "."), A(e), h('Each child in a list should have a unique "key" prop.%s%s See https://reactjs.org/link/warning-keys for more information.', t, a), A(null);
      }
    }
    function Oe(e, r) {
      {
        if (typeof e != "object")
          return;
        if (K(e))
          for (var t = 0; t < e.length; t++) {
            var a = e[t];
            ee(a) && je(a, r);
          }
        else if (ee(e))
          e._store && (e._store.validated = !0);
        else if (e) {
          var l = $e(e);
          if (typeof l == "function" && l !== e.entries)
            for (var s = l.call(e), c; !(c = s.next()).done; )
              ee(c.value) && je(c.value, r);
        }
      }
    }
    function hr(e) {
      {
        var r = e.type;
        if (r == null || typeof r == "string")
          return;
        var t;
        if (typeof r == "function")
          t = r.propTypes;
        else if (typeof r == "object" && (r.$$typeof === g || // Note: Memo only checks outer props here.
        // Inner props are checked in the reconciler.
        r.$$typeof === _))
          t = r.propTypes;
        else
          return;
        if (t) {
          var a = x(r);
          nr(t, e.props, "prop", a, e);
        } else if (r.PropTypes !== void 0 && !Q) {
          Q = !0;
          var l = x(r);
          h("Component %s declared `PropTypes` instead of `propTypes`. Did you misspell the property assignment?", l || "Unknown");
        }
        typeof r.getDefaultProps == "function" && !r.getDefaultProps.isReactClassApproved && h("getDefaultProps is only used on classic React.createClass definitions. Use a static property named `defaultProps` instead.");
      }
    }
    function mr(e) {
      {
        for (var r = Object.keys(e.props), t = 0; t < r.length; t++) {
          var a = r[t];
          if (a !== "children" && a !== "key") {
            A(e), h("Invalid prop `%s` supplied to `React.Fragment`. React.Fragment can only have `key` and `children` props.", a), A(null);
            break;
          }
        }
        e.ref !== null && (A(e), h("Invalid attribute `ref` supplied to `React.Fragment`."), A(null));
      }
    }
    function Fe(e, r, t, a, l, s) {
      {
        var c = Ke(e);
        if (!c) {
          var i = "";
          (e === void 0 || typeof e == "object" && e !== null && Object.keys(e).length === 0) && (i += " You likely forgot to export your component from the file it's defined in, or you might have mixed up default and named imports.");
          var m = Er(l);
          m ? i += m : i += Te();
          var d;
          e === null ? d = "null" : K(e) ? d = "array" : e !== void 0 && e.$$typeof === f ? (d = "<" + (x(e.type) || "Unknown") + " />", i = " Did you accidentally export a JSX literal instead of a component?") : d = typeof e, h("React.jsx: type is invalid -- expected a string (for built-in components) or a class/function (for composite components) but got: %s.%s", d, i);
        }
        var v = pr(e, r, t, l, s);
        if (v == null)
          return v;
        if (c) {
          var R = r.children;
          if (R !== void 0)
            if (a)
              if (K(R)) {
                for (var D = 0; D < R.length; D++)
                  Oe(R[D], e);
                Object.freeze && Object.freeze(R);
              } else
                h("React.jsx: Static children should always be an array. You are likely explicitly calling React.jsxs or React.jsxDEV. Use the Babel transform instead.");
            else
              Oe(R, e);
        }
        return e === p ? mr(v) : hr(v), v;
      }
    }
    function yr(e, r, t) {
      return Fe(e, r, t, !0);
    }
    function br(e, r, t) {
      return Fe(e, r, t, !1);
    }
    var wr = br, Cr = yr;
    I.Fragment = p, I.jsx = wr, I.jsxs = Cr;
  }()), I;
}
process.env.NODE_ENV === "production" ? ie.exports = kr() : ie.exports = Rr();
var O = ie.exports;
const B = "normal", Be = {
  type: B,
  title: void 0,
  children: void 0
}, _r = (n) => {
  if (z(n) || Array.isArray(n) && z(n == null ? void 0 : n.at(0)))
    return {
      type: B,
      title: void 0,
      content: n
    };
  const f = Array.isArray(n) ? n.at(0) : n, y = "\\[!([^\\n]+)\\](?:[ ]+([^\\n]+))?", p = new RegExp(y), b = f.match(p);
  if (!b)
    return {
      type: B,
      title: void 0,
      content: f
    };
  const [, k, w] = b;
  return {
    type: k,
    title: Array.isArray(n) ? n.at(1) : w,
    content: Array.isArray(n) ? [n.at(2).trim(), ...n.slice(3)] : f.replace(p, "").trim()
  };
}, xr = (n) => {
  if (!Array.isArray(n) || n.every((E) => !z(E)))
    return Be;
  const f = n.findIndex((E) => z(E)), y = n.at(f).props.children, { type: p, title: b, content: k } = _r(y), w = {
    ...n.at(f),
    props: { children: k }
  };
  return {
    type: p,
    title: b,
    children: [
      ...n.slice(0, f),
      w,
      ...n.slice(f + 1)
    ]
  };
}, ce = (n) => /* @__PURE__ */ o.createElement("svg", { xmlns: "http://www.w3.org/2000/svg", width: 24, height: 24, viewBox: "0 0 24 24", fill: "none", stroke: "currentColor", strokeWidth: 2, strokeLinecap: "round", strokeLinejoin: "round", className: "svg-icon lucide-pencil", ...n }, /* @__PURE__ */ o.createElement("path", { d: "M17 3a2.85 2.83 0 1 1 4 4L7.5 20.5 2 22l1.5-5.5Z" }), /* @__PURE__ */ o.createElement("path", { d: "m15 5 4 4" })), re = (n) => /* @__PURE__ */ o.createElement("svg", { xmlns: "http://www.w3.org/2000/svg", width: 24, height: 24, viewBox: "0 0 24 24", fill: "none", stroke: "currentColor", strokeWidth: 2, strokeLinecap: "round", strokeLinejoin: "round", className: "svg-icon lucide-clipboard-list", ...n }, /* @__PURE__ */ o.createElement("rect", { x: 8, y: 2, width: 8, height: 4, rx: 1, ry: 1 }), /* @__PURE__ */ o.createElement("path", { d: "M16 4h2a2 2 0 0 1 2 2v14a2 2 0 0 1-2 2H6a2 2 0 0 1-2-2V6a2 2 0 0 1 2-2h2" }), /* @__PURE__ */ o.createElement("path", { d: "M12 11h4" }), /* @__PURE__ */ o.createElement("path", { d: "M12 16h4" }), /* @__PURE__ */ o.createElement("path", { d: "M8 11h.01" }), /* @__PURE__ */ o.createElement("path", { d: "M8 16h.01" })), Le = (n) => /* @__PURE__ */ o.createElement("svg", { xmlns: "http://www.w3.org/2000/svg", width: 24, height: 24, viewBox: "0 0 24 24", fill: "none", stroke: "currentColor", strokeWidth: 2, strokeLinecap: "round", strokeLinejoin: "round", className: "svg-icon lucide-info", ...n }, /* @__PURE__ */ o.createElement("circle", { cx: 12, cy: 12, r: 10 }), /* @__PURE__ */ o.createElement("path", { d: "M12 16v-4" }), /* @__PURE__ */ o.createElement("path", { d: "M12 8h.01" })), te = (n) => /* @__PURE__ */ o.createElement("svg", { xmlns: "http://www.w3.org/2000/svg", width: 24, height: 24, viewBox: "0 0 24 24", fill: "none", stroke: "currentColor", strokeWidth: 2, strokeLinecap: "round", strokeLinejoin: "round", className: "svg-icon lucide-flame", ...n }, /* @__PURE__ */ o.createElement("path", { d: "M8.5 14.5A2.5 2.5 0 0 0 11 12c0-1.38-.5-2-1-3-1.072-2.143-.224-4.054 2-6 .5 2.5 2 4.9 4 6.5 2 1.6 3 3.5 3 5.5a7 7 0 1 1-14 0c0-1.153.433-2.294 1-3a2.5 2.5 0 0 0 2.5 2.5z" })), ne = (n) => /* @__PURE__ */ o.createElement("svg", { xmlns: "http://www.w3.org/2000/svg", width: 24, height: 24, viewBox: "0 0 24 24", fill: "none", stroke: "currentColor", strokeWidth: 2, strokeLinecap: "round", strokeLinejoin: "round", className: "svg-icon lucide-check", ...n }, /* @__PURE__ */ o.createElement("path", { d: "M20 6 9 17l-5-5" })), oe = (n) => /* @__PURE__ */ o.createElement("svg", { xmlns: "http://www.w3.org/2000/svg", width: 24, height: 24, viewBox: "0 0 24 24", fill: "none", stroke: "currentColor", strokeWidth: 2, strokeLinecap: "round", strokeLinejoin: "round", className: "svg-icon lucide-help-circle", ...n }, /* @__PURE__ */ o.createElement("circle", { cx: 12, cy: 12, r: 10 }), /* @__PURE__ */ o.createElement("path", { d: "M9.09 9a3 3 0 0 1 5.83 1c0 2-3 3-3 3" }), /* @__PURE__ */ o.createElement("path", { d: "M12 17h.01" })), ae = (n) => /* @__PURE__ */ o.createElement("svg", { xmlns: "http://www.w3.org/2000/svg", width: 24, height: 24, viewBox: "0 0 24 24", fill: "none", stroke: "currentColor", strokeWidth: 2, strokeLinecap: "round", strokeLinejoin: "round", className: "svg-icon lucide-alert-triangle", ...n }, /* @__PURE__ */ o.createElement("path", { d: "m21.73 18-8-14a2 2 0 0 0-3.48 0l-8 14A2 2 0 0 0 4 21h16a2 2 0 0 0 1.73-3Z" }), /* @__PURE__ */ o.createElement("path", { d: "M12 9v4" }), /* @__PURE__ */ o.createElement("path", { d: "M12 17h.01" })), Me = (n) => /* @__PURE__ */ o.createElement("svg", { xmlns: "http://www.w3.org/2000/svg", width: 24, height: 24, viewBox: "0 0 24 24", fill: "none", stroke: "currentColor", strokeWidth: 2, strokeLinecap: "round", strokeLinejoin: "round", className: "svg-icon lucide-zap", ...n }, /* @__PURE__ */ o.createElement("polygon", { points: "13 2 3 14 12 14 11 22 21 10 12 10 13 2" })), Tr = (n) => /* @__PURE__ */ o.createElement("svg", { xmlns: "http://www.w3.org/2000/svg", width: 24, height: 24, viewBox: "0 0 24 24", fill: "none", stroke: "currentColor", strokeWidth: 2, strokeLinecap: "round", strokeLinejoin: "round", className: "svg-icon lucide-bug", ...n }, /* @__PURE__ */ o.createElement("path", { d: "m8 2 1.88 1.88" }), /* @__PURE__ */ o.createElement("path", { d: "M14.12 3.88 16 2" }), /* @__PURE__ */ o.createElement("path", { d: "M9 7.13v-1a3.003 3.003 0 1 1 6 0v1" }), /* @__PURE__ */ o.createElement("path", { d: "M12 20c-3.3 0-6-2.7-6-6v-3a4 4 0 0 1 4-4h4a4 4 0 0 1 4 4v3c0 3.3-2.7 6-6 6" }), /* @__PURE__ */ o.createElement("path", { d: "M12 20v-9" }), /* @__PURE__ */ o.createElement("path", { d: "M6.53 9C4.6 8.8 3 7.1 3 5" }), /* @__PURE__ */ o.createElement("path", { d: "M6 13H2" }), /* @__PURE__ */ o.createElement("path", { d: "M3 21c0-2.1 1.7-3.9 3.8-4" }), /* @__PURE__ */ o.createElement("path", { d: "M20.97 5c0 2.1-1.6 3.8-3.5 4" }), /* @__PURE__ */ o.createElement("path", { d: "M22 13h-4" }), /* @__PURE__ */ o.createElement("path", { d: "M17.2 17c2.1.1 3.8 1.9 3.8 4" })), Sr = (n) => /* @__PURE__ */ o.createElement("svg", { xmlns: "http://www.w3.org/2000/svg", width: 24, height: 24, viewBox: "0 0 24 24", fill: "none", stroke: "currentColor", strokeWidth: 2, strokeLinecap: "round", strokeLinejoin: "round", className: "svg-icon lucide-list", ...n }, /* @__PURE__ */ o.createElement("line", { x1: 8, y1: 6, x2: 21, y2: 6 }), /* @__PURE__ */ o.createElement("line", { x1: 8, y1: 12, x2: 21, y2: 12 }), /* @__PURE__ */ o.createElement("line", { x1: 8, y1: 18, x2: 21, y2: 18 }), /* @__PURE__ */ o.createElement("line", { x1: 3, y1: 6, x2: 3.01, y2: 6 }), /* @__PURE__ */ o.createElement("line", { x1: 3, y1: 12, x2: 3.01, y2: 12 }), /* @__PURE__ */ o.createElement("line", { x1: 3, y1: 18, x2: 3.01, y2: 18 })), We = (n) => /* @__PURE__ */ o.createElement("svg", { xmlns: "http://www.w3.org/2000/svg", width: 24, height: 24, viewBox: "0 0 24 24", fill: "none", stroke: "currentColor", strokeWidth: 2, strokeLinecap: "round", strokeLinejoin: "round", className: "svg-icon lucide-quote", ...n }, /* @__PURE__ */ o.createElement("path", { d: "M3 21c3 0 7-1 7-8V5c0-1.25-.756-2.017-2-2H4c-1.25 0-2 .75-2 1.972V11c0 1.25.75 2 2 2 1 0 1 0 1 1v1c0 1-1 2-2 2s-1 .008-1 1.031V20c0 1 0 1 1 1z" }), /* @__PURE__ */ o.createElement("path", { d: "M15 21c3 0 7-1 7-8V5c0-1.25-.757-2.017-2-2h-4c-1.25 0-2 .75-2 1.972V11c0 1.25.75 2 2 2h.75c0 2.25.25 4-2.75 4v3c0 1 0 1 1 1z" })), jr = {
  note: {
    icon: ce,
    color: "#306BD6",
    backgroundColor: "#E6ECF7"
  },
  abstract: {
    icon: re,
    color: "#56BCBB",
    backgroundColor: "#EEF8F8"
  },
  summary: {
    icon: re,
    color: "#56BCBB",
    backgroundColor: "#EEF8F8"
  },
  tldr: {
    icon: re,
    color: "#56BCBB",
    backgroundColor: "#EEF8F8"
  },
  info: {
    icon: Le,
    color: "#306BD6",
    backgroundColor: "#E6ECF7"
  },
  todo: {
    icon: Le,
    color: "#306BD6",
    backgroundColor: "#E6ECF7"
  },
  tip: {
    icon: te,
    color: "#56BCBB",
    backgroundColor: "#EEF8F8"
  },
  hint: {
    icon: te,
    color: "#56BCBB",
    backgroundColor: "#EEF8F8"
  },
  important: {
    icon: te,
    color: "#56BCBB",
    backgroundColor: "#EEF8F8"
  },
  success: {
    icon: ne,
    color: "#5AC561",
    backgroundColor: "#E2F1E7"
  },
  check: {
    icon: ne,
    color: "#5AC561",
    backgroundColor: "#E2F1E7"
  },
  done: {
    icon: ne,
    color: "#5AC561",
    backgroundColor: "#E2F1E7"
  },
  question: {
    icon: oe,
    color: "#85DA47",
    backgroundColor: "#EAF5E2"
  },
  help: {
    icon: oe,
    color: "#85DA47",
    backgroundColor: "#EAF5E2"
  },
  faq: {
    icon: oe,
    color: "#85DA47",
    backgroundColor: "#EAF5E2"
  },
  warning: {
    icon: ae,
    color: "#F09637",
    backgroundColor: "#F6EDE0"
  },
  caution: {
    icon: ae,
    color: "#F09637",
    backgroundColor: "#F6EDE0"
  },
  attention: {
    icon: ae,
    color: "#F09637",
    backgroundColor: "#F6EDE0"
  },
  danger: {
    icon: Me,
    color: "#EA394B",
    backgroundColor: "#F4E1E5"
  },
  error: {
    icon: Me,
    color: "#EA394B",
    backgroundColor: "#F4E1E5"
  },
  bug: {
    icon: Tr,
    color: "#E1315A",
    backgroundColor: "#F3DFE7"
  },
  example: {
    icon: Sr,
    color: "#7952F6",
    backgroundColor: "#EAE6F7"
  },
  quote: {
    icon: We,
    color: "#9E9E9E",
    backgroundColor: "#EEEEEE"
  },
  cite: {
    icon: We,
    color: "#9E9E9E",
    backgroundColor: "#EEEEEE"
  },
  [B]: {
    icon: ce,
    color: "#5688F7",
    backgroundColor: "#E6ECF7"
  }
}, u = ({
  type: n,
  title: f,
  options: y,
  children: p
}) => {
  const { icon: b, color: k, backgroundColor: w } = { ...jr, ...y }[n], E = typeof b == "function" ? b : ce;
  return /* @__PURE__ */ O.jsxs("div", { style: { backgroundColor: w, padding: "1rem 1rem 0.625rem 1.5rem", borderRadius: "0.5rem", margin: "0.5rem 0" }, children: [
    /* @__PURE__ */ O.jsxs("div", { style: { color: k, display: "flex", gap: "0.25rem", fontWeight: 600 }, children: [
      /* @__PURE__ */ O.jsx(E, { width: 20 }),
      /* @__PURE__ */ O.jsx("p", { children: f ?? n.charAt(0).toUpperCase() + n.slice(1) })
    ] }),
    /* @__PURE__ */ O.jsx("div", { style: { padding: "1rem 0" }, children: p })
  ] });
}, Ve = {
  note: u,
  abstract: u,
  summary: u,
  tldr: u,
  info: u,
  todo: u,
  tip: u,
  hint: u,
  important: u,
  success: u,
  check: u,
  done: u,
  question: u,
  help: u,
  faq: u,
  warning: u,
  caution: u,
  attention: u,
  danger: u,
  error: u,
  bug: u,
  example: u,
  quote: u,
  cite: u,
  [B]: u
}, Ie = (n, f, y) => !!(Object.keys(Ve).includes(n) || f && Object.keys(f).includes(n) || y && Object.keys(y).includes(n)), Or = ({
  components: n,
  options: f,
  ...y
}) => {
  const { type: p, title: b, children: k } = xr(y.children), w = Ie(p, n, f) ? p : B, E = { ...Ve, ...n }[w] || u;
  return /* @__PURE__ */ O.jsx(
    E,
    {
      type: w,
      title: Ie(p) ? b : p,
      options: f,
      children: k
    }
  );
}, Pr = ({ ...n }) => /* @__PURE__ */ O.jsx(Or, { ...n });
export {
  Pr as ObsidianCallout
};
